packages/fxa-shared/package.json

```diff
@@ -5,7 +5,51 @@
   "main": "dist/index.js",
   "exports": {
     ".": "./dist/index.js",
-    "./": "./dist/"
+    "./*": "./dist/*",
+    "./auth": "./dist/auth/index.js",
+    "./auth/*": "./dist/auth/*.js",
+    "./cache/*": "./dist/cache/*.js",
+    "./configuration/*": "./dist/configuration/*.js",
+    "./connected-services": "./dist/connected-services/index.js",
+    "./connected-services/*": "./dist/connected-services/*.js",
+    "./coverage/*": "./dist/coverage/*.js",
+    "./db": "./dist/db/index.js",
+    "./db/*": "./dist/db/*.js",
+    "./db/config": "./dist/db/config.js",
+    "./db/models": "./dist/db/models/index.js",
+    "./db/models/auth": "./dist/db/models/auth/index.js",
+    "./db/models/auth/*": "./dist/db/models/auth/*.js",
+    "./db/models/profile": "./dist/db/models/profile/index.js",
+    "./db/mysql": "./dist/db/mysql.js",
+    "./db/redis": "./dist/db/redis.js",
+    "./dto/*": "./dist/dto/*.js",
+    "./email/*": "./dist/email/*.js",
+    "./experiments/*": "./dist/experiments/*.js",
+    "./express": "./dist/express/index.js",
+    "./express/*": "./dist/express/*.js",
+    "./feature-flags": "./dist/feature-flags/index.js",
+    "./feature-flags/*": "./dist/feature-flags/*.js",
+    "./guards": "./dist/guards/index.js",
+    "./guards/*": "./dist/guards/*.js",
+    "./l10n/*": "./dist/l10n/*.js",
+    "./lib/*": "./dist/lib/*.js",
+    "./log": "./dist/log/index.js",
+    "./log/*": "./dist/log/*.js",
+    "./metrics/*": "./dist/metrics/*.js",
+    "./nestjs/*": "./dist/nestjs/*.js",
+    "./oauth/*": "./dist/oauth/*.js",
+    "./payments/*": "./dist/payments/*.js",
+    "./payments/configuration/*": "./dist/payments/configuration/*.js",
+    "./payments/iap/*": "./dist/payments/iap/*.js",
+    "./payments/iap/apple-app-store/types": "./dist/payments/iap/apple-app-store/types/index.js",
+    "./payments/iap/google-play/types": "./dist/payments/iap/google-play/types/index.js",
+    "./payments/stripe": "./dist/payments/stripe.js",
+    "./payments/stripe-firestore": "./dist/payments/stripe-firestore.js",
+    "./scripts/*": "./dist/scripts/*.js",
+    "./sentry": "./dist/sentry/index.js",
+    "./sentry/*": "./dist/sentry/*.js",
+    "./subscriptions/*": "./dist/subscriptions/*.js",
+    "./tracing/*": "./dist/tracing/*.js"
```

Comment:
Despite not upgrading anything, my local environment started throwing module not found errors that are fixed by populating this list.  It is a bit unwieldy because a/b/c can be a/b/c/index.js or a/b/c.js so it must specific when that ambiguity would case an error.

Reply:
What started throwing the errors? Was a specific server? I am curious to see if I can reproduce.

Reply:
I _think_ it was any TS importing something from fxa-shared.  Here's an example

```
 162   │ ➤ YN0000: [fxa-auth-server]: /Users/barrychen/code/fxa/node_modules/esbuild-register/dist/node.js:2708
 163   │ ➤ YN0000: [fxa-auth-server]:         throw error;
 164   │ ➤ YN0000: [fxa-auth-server]:         ^
 165   │ ➤ YN0000: [fxa-auth-server]: 
 166   │ ➤ YN0000: [fxa-auth-server]: Error: Cannot find module '/Users/barrychen/code/fxa/node_modules/fxa-shared/dist/db/config'
 167   │ ➤ YN0000: [fxa-auth-server]:     at createEsmNotFoundErr (node:internal/modules/cjs/loader:960:15)
 168   │ ➤ YN0000: [fxa-auth-server]:     at finalizeEsmResolution (node:internal/modules/cjs/loader:953:15)
 169   │ ➤ YN0000: [fxa-auth-server]:     at resolveExports (node:internal/modules/cjs/loader:482:14)
 170   │ ➤ YN0000: [fxa-auth-server]:     at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
 171   │ ➤ YN0000: [fxa-auth-server]:     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
 172   │ ➤ YN0000: [fxa-auth-server]:     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
 173   │ ➤ YN0000: [fxa-auth-server]:     at Module.require (node:internal/modules/cjs/loader:1005:19)
 174   │ ➤ YN0000: [fxa-auth-server]:     at require (node:internal/modules/cjs/helpers:102:18)
 175   │ ➤ YN0000: [fxa-auth-server]:     at Object.<anonymous> (/Users/barrychen/code/fxa/packages/fxa-auth-server/config/index.ts:6:50)
 176   │ ➤ YN0000: [fxa-auth-server]:     at Module._compile (node:internal/modules/cjs/loader:1105:14) {
 177   │ ➤ YN0000: [fxa-auth-server]:   code: 'MODULE_NOT_FOUND',
 178   │ ➤ YN0000: [fxa-auth-server]:   path: '/Users/barrychen/code/fxa/node_modules/fxa-shared/package.json'
 179   │ ➤ YN0000: [fxa-auth-server]: }
```

Reply:
Okay, I am reproducing this behavior. I can't seem to figure out what change triggered this either, but I think this is actually the correct way to setup the package file anyways. I've see warnings about the line `"./": "./dist/"` not in both vscode's IDE, and it was output at the beginning of certain tests too.